src/tribler/core/utilities/pony_utils.py

```diff
@@ -281,27 +378,71 @@ def acquire_lock(self):
             info.lock_wait_total_duration += lock_wait_duration
 
     def release_lock(self):
+        # Adds tracking of a db_session's total lock hold duration
         super().release_lock()
         info = local.db_session_info
         if info is not None:
             acquire_time = self._acquire_time
             lock_hold_duration = time.time() - acquire_time
             info.lock_hold_total_duration += lock_hold_duration
 
+    def set_transaction_mode(self, connection, cache):
+        with marking_corrupted_db(self.pool.filename):
+            return super().set_transaction_mode(connection, cache)
+
+    def execute(self, cursor, sql, arguments=None, returning_id=False):
+        with marking_corrupted_db(self.pool.filename):
+            return super().execute(cursor, sql, arguments, returning_id)
+
+    def mark_db_as_malformed(self):
+        filename = self.pool.filename
+        if not Path(filename).exists():
+            raise RuntimeError(f'Corrupted database file not found: {filename!r}')
+
+        marker_filename = filename + '.is_corrupted'
+        Path(marker_filename).touch()
+
+    def get_pool(self, is_shared_memory_db, filename, create_db=False, **kwargs):
+        if not (is_shared_memory_db or filename == ':memory:'):
+            filename = absolutize_path(filename, frame_depth=cut_traceback_depth+5)  # see the base method for details
+            handle_db_if_corrupted(filename)
+        return TriblerPool(is_shared_memory_db, filename, create_db, **kwargs)
+
+
+class TriblerPool(SQLitePool):
```

Comment:
Please, add a description to the class 🙏

Reply:
Added